Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1796 @@
+
+tar xzf apache-tomcat-7.0.86.tar.gz && rm -f apache-tomcat-7.0.86.tar.gz
+
+ln -s /usr/share/apache-tomcat-7.0.86 /usr/share/tomcat7-blzg
+
+useradd -m -d /var/bigdata -s /bin/false blazegraph
+
+cat > /var/bigdata/.bash_profile <<END_BG
+export BLZG_CONF=/etc/bigdata
+export CATALINA_HOME=/usr/share/tomcat7-blzg
+export CATALINA_PID="/usr/share/tomcat7-blzg/catalina.pid"
+export JAVA_HOME=/usr/lib/jvm/java-8-oracle/jre
+export BLZG_USER=blazegraph
+export PATH=/usr/lib/jvm/java-8-oracle/jre/bin:/usr/share/tomcat7-blzg/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin
+# - adjust for your available memory
+#export JAVA_OPTS="-server -Djava.awt.headless=true -Xmx8g -Xms8g -Dcom.bigdata.rdf.sail.webapp.ConfigParams.propertyFile=/etc/bigdata/RWStore.properties -Dlog4j.configuration=/etc/bigdata/log4j.properties -Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8 -XX:+UseG1GC -XX:+DisableExplicitGC"
+export JAVA_OPTS="-server -Djava.awt.headless=true -Xmx128m -Xms128m -Dcom.bigdata.rdf.sail.webapp.ConfigParams.propertyFile=/etc/bigdata/RWStore.properties -Dlog4j.configuration=/etc/bigdata/log4j.properties -Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8 -XX:+UseG1GC -XX:+DisableExplicitGC"
+END_BG
+
+chown blazegraph:blazegraph /var/bigdata/.bash_profile
+```
+Edit `/usr/share/tomcat7-blzg/conf/server.xml` to set the service ports to something other than the Tomcat defaults, so as not to conflict with the Fedora Tomcat.
+```
+cd /usr/share/tomcat7-blzg/conf
+
+cp server.xml server.xml.bak
+
+sed -i "s|port=\"8005|port=\"8006|g" server.xml
+
+sed -i "s|port=\"8080|port=\"8081|g" server.xml
+
+sed -i "s|port=\"8009|port=\"8010|g" server.xml
+
+sed -i "s|redirectPort=\"8443|redirectPort=\"8444|g" server.xml
+```
+Install and configure Blazegraph and its service.
+```
+mkdir -p /var/bigdata/logs
+
+mkdir /etc/bigdata
+
+cd ~
+
+git clone https://github.com/discoverygarden/blazegraph_conf
+
+cp blazegraph_conf/RWStore.properties /etc/bigdata/
+
+cp blazegraph_conf/log4j.properties /etc/bigdata/
+
+cp blazegraph_conf/blazegraph_init /etc/init.d/blazegraph
+
+rm -rf blazegraph_conf
+
+cd /usr/share/tomcat7-blzg/webapps/
+
+wget https://sourceforge.net/projects/bigdata/files/bigdata/2.1.4/blazegraph.war/download -O blazegraph.war
+
+chown -R blazegraph:blazegraph /usr/share/apache-tomcat-7.0.86 /var/bigdata/ /etc/bigdata/
+
+chmod +x /etc/init.d/blazegraph
+
+update-rc.d -f blazegraph remove
+
+update-rc.d blazegraph defaults
+
+systemctl start blazegraph
+```
+Check that Blazegraph is present at port 8081.
+
+```
+systemctl stop blazegraph
+```
+
+
+#### Fits  <a id="fits"></a>
+```
+cd /usr/local
+
+wget http://projects.iq.harvard.edu/files/fits/files/fits-0.10.1.zip
+
+unzip -o fits-0.10.1.zip && rm -rf fits-0.10.1.zip && ln -s fits-0.10.1 fits && chmod a+x /usr/local/fits/fits.sh
+```
+
+#### Adore-Djatoka Install<a id="adore-djatoka-install"></a>
```

Comment:
If we get rid of djatoka we should be able to use the openjdk which makes maintenance a lot easier.